src/jobs/job_stack.rs

```diff
@@ -45,7 +48,16 @@ impl JobStack {
                         }
                     }
                     None => {
-                        return Ok(None);
+                        if let JobRef::PreviousOrQuit = job_ref {
+                            return Err(None);
+                        }
+
+                        // this has a side effect of reloading the job, which isn't ideal
+                        // but the main loop needs some mission to avoid spinning forever
```

Comment:
This is the main caveat for this implementation; every time I tried to make it avoid reloading here, the result was the main `loop` constantly trying to pick a job and never running any mission, which freezes the application (can't even quit with e.g. ctrl-c and need to kill from another session).

If you can think of a way to make it so this _doesn't_ reload the same job again that would be great!